### ISSUE TITLE
Add todo app to examples

### ISSUE THREAD

**Issue by tbosch** [angular/angular#604]
@davideast already has a running app locally.

**Comment by gdi2290** [comment]
is there a list of examples that needs to be built?

**Comment by mhevery** [comment]
This is for the @davideast example: https://github.com/davideast/ng2do

**Comment by mhevery** [comment]
@gdi2290 do you have some time to work on this? Are you still interested?

**Comment by gdi2290** [comment]
@mhevery yup I'm still interested, I need to update my refactor then I can open a PR. To get the filters working I would need to update zone.js to watch location hash or manually trigger the change detection. Alternatively, I could filter the collection and `.preventDefault()` when the user clicks on the link, but I rather update zone.js
![screen shot 2015-02-24 at 11 39 13 am](https://cloud.githubusercontent.com/assets/1016365/6357495/ce2831bc-bc19-11e4-994d-6086de128dac.png)
```html
<ul [style]="styles.filters">
    <li>
      <a href="#/" [class.selected]="locationPath == ''">All</a>
    </li>
    <li>
      <a href="#/active" [class.selected]="locationPath == 'active'">Active</a>
    </li>
    <li>
      <a href="#/completed" [class.selected]="locationPath == 'completed'">Completed</a>
    </li>
  </ul>
```

**Comment by mhevery** [comment]
@btford What do you think on this: https://github.com/angular/angular/issues/604#issuecomment-75830601

**Comment by davideast** [comment]
@gdi2290 I'm going to have my example in a PR soon. I think it would be best to work off of that example. Once I get the example merged in I can review and PRs from you.

Also if you have any ideas for other examples please let us know!

**Comment by gdi2290** [comment]
@mhevery @davideast I got everything working now. What I used `.preventDefault()` for each link and have the initial state set by the component [/todo-footer.js#L22](https://github.com/gdi2290/angular2do/blob/acb0f930e9a9c268584f6d1c93be331bdaa1b8aa/app/components/todo-footer/todo-footer.js#L22) until a location service is made or I add a router. I need some opinions on how the code should be organized since I can move logic around to fit any style. For example I'm storing the state and business logic of todos/filter/editing in the [TodoStore](https://github.com/gdi2290/angular2do/blob/acb0f930e9a9c268584f6d1c93be331bdaa1b8aa/app/stores/TodoStore.js). A small refactor could switch the design into the glorified flux architecture which is also known as more as pub/sub architecture. I could more todo logic in the app component and pass state down to it's children for that unidirectional whole app rerender mindset. I could make a mvp version which is a lot less code (one component). One sidenote was the css, which wasn't built with components in mind so I had to rip it up and reassemble it in the right place. I can make an example of each one or each example just not a todo app

Todo: 
location or router service [todo-footer.js#L21](https://github.com/gdi2290/angular2do/blob/acb0f930e9a9c268584f6d1c93be331bdaa1b8aa/app/components/todo-footer/todo-footer.js#L21)
pluralize service [todo-footer.js#L32](https://github.com/gdi2290/angular2do/blob/acb0f930e9a9c268584f6d1c93be331bdaa1b8aa/app/components/todo-footer/todo-footer.js#L32)
filter foreach [todo-main.js#L30]
(https://github.com/gdi2290/angular2do/blob/acb0f930e9a9c268584f6d1c93be331bdaa1b8aa/app/components/todo-main/todo-main.js#L30)

repo:
https://github.com/gdi2290/angular2do
live view:
http://gdi2290.net/angular2do/

component overview
```html
<todo-app>
  <h1></h2>
  <todo-header>
    <input type="checkbox">
    <input type="input">
  </todo-header>
  <todo-main>
    <ul>
      <li foreach="todo in todos">
        <input type="checkbox">
        <todo-item><todo-item>
    </ul>
  </todo-main>
  <todo-footer></todo-footer>
  <footer></footer>
</todo-app>
```

**Comment by mhevery** [comment]
@davideast  Can you comment on this. 

- Are we planning to have this as a separate repo or part of Angular? I would argue as part of angular so that we can have it run a end to end tests and verify that the app works, and so that we can keep it in sync with all of the changes.
- If above true, can you send a PR? I think it should be in here: https://github.com/angular/angular/tree/master/modules/examples

**Comment by pkozlowski-opensource** [comment]
Resolved in afda43dc0281e3671d6df3218749d5050d47f188
